### PR TITLE
Adding minimal logic to Reconcile method and setting up Scheme for future use

### PR DESCRIPTION
**Pull request by sadasu**
Another step towards a minimally functioning operator to be deployed by CVO.

### CODE REVIEW

**Comment by dhellmann** [comment]
/assign @dhellmann 

Let's get https://github.com/openshift/release/pull/11500 in place before we merge this.

File: controllers/provisioning_controller.go [hunk 1]
```diff
@@ -22,15 +22,32 @@ import (
...
+
 // ProvisioningReconciler reconciles a Provisioning object
 type ProvisioningReconciler struct {
-	client.Client
-	Log    logr.Logger
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	Client client.Client
 	Scheme *runtime.Scheme
+	Log    logr.Logger
```
Comment (dhellmann) [inline controllers/provisioning_controller.go:42]: Does the logger need to be public? Is anything outside of the reconciler going to use it?

File: controllers/provisioning_controller.go [hunk 2]
```diff
@@ -39,10 +56,29 @@ type ProvisioningReconciler struct {
...
-	_ = r.Log.WithValues("provisioning", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("provisioning", req.NamespacedName)
+
+	infra := &osconfigv1.Infrastructure{}
+	err := r.Client.Get(ctx, client.ObjectKey{
+		Name: "cluster",
+	}, infra)
+	if err != nil {
+		log.Info("Unable to determine Platform that the Operator is running on.")
```
Comment (dhellmann) [inline controllers/provisioning_controller.go:67]: This log message should include the error that caused the failure.

File: controllers/provisioning_controller.go [hunk 3]
```diff
@@ -39,10 +56,29 @@ type ProvisioningReconciler struct {
...
+		return ctrl.Result{}, err
+	}
 
-	// your logic here
+	// Disable ourselves on platforms other than bare metal
+	if infra.Status.Platform != osconfigv1.BareMetalPlatformType {
+		log.Info("Not Baremetal Platform. Setting CBO to disabled state.")
```
Comment (dhellmann) [inline controllers/provisioning_controller.go:73]: This log message should include the value of the platform. If you log before testing, then the log message on line 81 isn't needed.

File: main.go [hunk 4]
```diff
@@ -74,6 +74,18 @@ func main() {
 	}
 	// +kubebuilder:scaffold:builder
 
+	setupLog.Info("Registering Components.")
```
Comment (dhellmann) [inline main.go:77]: We should consider a style guide for log messages, so they're consistent. I notice this uses caps and punctuation, but line 89 doesn't. I like keeping them stripped down, myself, but could go either way. Thoughts?

File: controllers/provisioning_controller.go [hunk 5]
```diff
@@ -39,10 +56,29 @@ type ProvisioningReconciler struct {
...
+		Name: "cluster",
+	}, infra)
+	if err != nil {
+		log.Info("Unable to determine Platform that the Operator is running on.")
+		return ctrl.Result{}, err
+	}
 
-	// your logic here
+	// Disable ourselves on platforms other than bare metal
+	if infra.Status.Platform != osconfigv1.BareMetalPlatformType {
```
Comment (dhellmann) [inline controllers/provisioning_controller.go:72]: This behavior could be tested with a unit test. Let's add a basic test. I know kubebuilder is recommending ginkgo, but we can just use normal test functions and packages like we do in the other metal3 components for now and consider ginkgo later.

**Review by dhellmann — CHANGES_REQUESTED** [review]

File: controllers/provisioning_controller.go (hunk 5 above)
Comment (sadasu) [reply controllers/provisioning_controller.go:63]: Yes, please check the latest now.

File: controllers/provisioning_controller.go (hunk 1 above)
Comment (sadasu) [reply controllers/provisioning_controller.go:41]: It was put into Reconciler by kubebuilder. Just the formatting changed in this PR. I'll leave a TODO here for us to see if there is a reason why it needs to be part of the Reconciler struct.

File: controllers/provisioning_controller.go (hunk 3 above)
Comment (sadasu) [reply controllers/provisioning_controller.go:64]: Done.

File: main.go (hunk 4 above)
Comment (sadasu) [reply main.go:83]: This was added by me and the one in line 89 was added by kubebuilder. I'll make it consistent with the one added by kubebuilder.

**Pull request merged**